### PR TITLE
expanded module functionality

### PR DESCRIPTION
**Pull request by chrismazanec**
- all changes backwards compatible
- cleanup (trailing spaces, formatting, removed unused entities)
- all resources created dynamically using `count` or `for_each` depending on module configuration
- added outputs descriptions
- added new variables/module options to better cover SSM preferences : `bucket_key_prefix`, `idle_session_timeout`, `run_as_default_user`, `shell_profile_windows`, `shell_profile_linux`, `enable_run_as`
- added new outputs: `iam_policy_arn`, `document_name`, `document_arn`
- updated `README.md` accordingly, added `Requirements`, `Providers` and `Notes` sections
- addressed https://github.com/bridgecrewio/terraform-aws-session-manager/issues/2 by removing unused code blocks

### CODE REVIEW

**Review by stephentan-luddart — APPROVED** [review]

**Comment by JamesWoolfenden** [comment]
can you update to the current master and ill review it again,

**Comment by apanzerj** [comment]
Hiya! Any update on this? I just need to be able to use the shell profiles logic. I made a PR here #17

**Comment by JamesWoolfenden** [comment]
i think most of this PR has made it one way or another to the module, if you sync with master we can see if theres anything else?

File: README.md [hunk 1]
```diff
@@ -29,62 +29,90 @@ module "ssm" {
...
+| vpc\_endpoint\_ec2messages | ["vpce-0f507468fb9b06b8b"] | VPC Endpoint for EC2 Messages |
+| vpc\_endpoint\_ssmmessages | ["vpce-0fe2cb670d40ec053"] | VPC Endpoint for SSM Messages |
+| vpc\_endpoint\_s3 | ["vpce-0a8ebde94fa301a4a"] | VPC Endpoint for S3 |
+| vpc\_endpoint\_logs | ["vpce-08c90d8df9ef37f90"] | VPC Endpoint for CloudWatch Logs |
+| vpc\_endpoint\_kms | ["vpce-07ddc11beac1d4a3f"] | VPC Endpoint for KMS |
 
 ## SSM Usage Example
 
-* Launch an instance using the ssm_profile created by Terraform
+* Launch an instance using the ssm\_profile created by Terraform
```
Comment (rohancragg) [inline README.md:115]: Could you clarify what you mean by 'launch an instance using a profile'?

Does this only work for newly launched instances?

What is meant by an Instance Profile?

Comment (apanzerj) [reply README.md:115]: It's been a while but my memory says that it refers to the instance profile that allows ssm actions for a given instance. You can apply that profile to existing insurances.

**Comment by chrismazanec** [comment]
hi all, I addressed part of this PR in a new one;
https://github.com/bridgecrewio/terraform-aws-session-manager/pull/20

**Pull request closed without merging**